openpype/plugins/publish/collect_input_representations_to_versions.py

```diff
@@ -23,25 +23,27 @@ def process(self, context):
         representations = set()
         for instance in context:
             inst_repre = instance.data.get("inputRepresentations", [])
-            representations.update(inst_repre)
+            if inst_repre:
+                representations.update(inst_repre)
 
         representations_docs = get_representations(
             project_name=context.data["projectEntity"]["name"],
             representation_ids=representations,
             fields=["_id", "parent"])
 
         representation_id_to_version_id = {
-            repre["_id"]: repre["parent"] for repre in representations_docs
+            str(repre["_id"]): repre["parent"]
+            for repre in representations_docs
         }
 
         for instance in context:
             inst_repre = instance.data.get("inputRepresentations", [])
             if not inst_repre:
                 continue
 
-            input_versions = instance.data.get("inputVersions", [])
+            instance.data.setdefault("inputVersions", [])
+            input_versions = instance.data["inputVersions"]
```

Comment:
```suggestion
            input_versions = instance.data.setdefault("inputVersions", [])
```

`dict.setdefault` already returns the value itself.